### PR TITLE
rust/src/smb/nbss_record-v1

### PR DESCRIPTION
**Pull request by jufajardini**
- add unit test for parse_nbss_record
- add unit test for parse_nbss_record_partial

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
- there isn't one, so far

Describe changes:
- nbss_records: add tag for unit tests to be run with cargo test
- nbss_records: add unit test for parse_nbss_record
- nbss_records: add unit test for parse_nbss_record_partial

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by jufajardini** [comment]
For now, the tests have only one packet as buffer, each. I wasn't able to find an NBSS pcap where message type wasn't session message (0x00).

Should I add more cases inside each test, or add more tests, for each different case? Should I manually create packets with different message types?

File: rust/src/smb/nbss_records.rs [hunk 1]
```diff
@@ -87,3 +87,77 @@ named!(pub parse_nbss_record_partial<NbssRecord>,
...
+        /* length */        0x00, 0x00, 0x29,
+        /* data < lenght*/  0xff, 0x53, 0x4d, 0x42, 0x04, 0x00, 0x00, 0x00, 
+                            0x00, 0x18, 0x43, 0xc8, 0x00, 0x00, 0x00, 0x00,
+                            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 
+                            0x02, 0x08, 0xbd, 0x20, 0x02, 0x08, 0x06, 0x00,
+                            0x02, 0x40, 0xff, 0xff, 0xff, 0xff, 0x00, 0x00];
+
+        let result = parse_nbss_record_partial(&buff);
+        match result {
+            Ok((_remainder,p)) => {
```
Comment (victorjulien) [inline rust/src/smb/nbss_records.rs:145]: can you add a test for the remainder as well?

File: rust/src/smb/nbss_records.rs [hunk 2]
```diff
@@ -87,3 +87,77 @@ named!(pub parse_nbss_record_partial<NbssRecord>,
...
+                           0x0a, 0x00, 0x01, 0x00, 0x04, 0x11, 0x00, 0x00, 0x00,
+                           0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0xfd, 0xe3,
+                           0x00, 0x80, 0x2a, 0x55, 0xc4, 0x38, 0x89, 0x03, 0xcd,
+                           0x01, 0x2c, 0x01, 0x00, 0x10, 0x00, 0xfe, 0x82, 0xf1,
+                           0x64, 0x0b, 0x66, 0xba, 0x4a, 0xbb, 0x81, 0xe1, 0xea,
+                           0x54, 0xae, 0xb8, 0x66];
+
+        let result = parse_nbss_record(&buff);
+        match result {
+            Ok((_remainder,p)) => {
```
Comment (victorjulien) [inline rust/src/smb/nbss_records.rs:114]: can you check for the data as well?

**Comment by jufajardini** [comment]
Closed with new PR: https://github.com/OISF/suricata/pull/5689

**Pull request closed without merging**